### ISSUE TITLE
GPX Support

### ISSUE THREAD

**Issue by valentine195** [valentine195/obsidian-leaflet-plugin#189]
I would like to display recorded gpx tracks (bicycle, car, camper, etc) on an obsidian-leafleat map. 

With leaflet I used the [leaflet-gpx plugin](https://github.com/mpetazzoni/leaflet-gpx) for this.

Could you implement the feature in obsidian-leaflet as well?

_Originally posted by @ramGHnet in https://github.com/valentine195/obsidian-leaflet-plugin/discussions/165_

**Comment by Moonbase59** [comment]
Would be a great addition (read my notes in the discussion), since one would save converting to/from GeoJSON (losing data) and only need to keep the GPX file.

Only GPX’ers tend to wish for a lot of data displayed (avg/min/max heart rate, blood pressure, speed, elevation up/down, distance travelled, etc.). I _think_ many could be displayed in the lower area of the map, selectable by a setting (like, a fixed set of properties, and one could switch each on/off?). Not all devices provide all data types.

The other question is how to display the track. Usually, one property (like speed, elevation, heart rate) is overlaid on the track by colorizing it. (See my MyTrails-creenhots. Here it’s speed that colorizes the track.)

**Comment by valentine195** [comment]
Basic gpx support added in 3.24.0. See [here](https://github.com/valentine195/obsidian-leaflet-plugin#gpx).

**Comment by Moonbase59** [comment]
How about adding `gpxColor:` (like the other ones), to be used as default if the track isn’t colorized by data?

**Comment by Moonbase59** [comment]
A GPX _waypoint_ currently

- shows the _Obsidian tooltip_ when hovering, and the `<name>…</name>` property,
- shows the _Leaflet_ tooltip when clicking it, displaying _both_ `<name>` (bolded) plus `<desc>` properties without a blank in between.

![Auswahl_108](https://user-images.githubusercontent.com/3706922/128597304-1e426840-bbe1-474a-ad82-45d1111f2cb9.png)

![Auswahl_109](https://user-images.githubusercontent.com/3706922/128597307-dbacf329-8487-4523-8ea6-459641a36ce6.png)

Suggest you use your type of tooltip instead, so as not to confuse users.

Waypoint data the aboce came from (using _MyTrails_ on Android):

```xml
<wpt lat="48.27693557739258" lon="10.317170143127441"><ele>500</ele><time>2021-07-29T15:20:14Z</time><name><![CDATA[29.7., 17:20]]></name><desc><![CDATA[bei 48,27694N, 10,31717O]]></desc><fix>3d</fix></wpt>
```

**Comment by valentine195** [comment]
Closed with 4.0.0 release.
